geoutils/georaster.py

```diff
@@ -822,10 +913,179 @@ def format_value(value):
+
+        :param xi: Indices (or coordinates) of x direction to check.
+        :param yj: Indices (or coordinates) of y direction to check.
+        :param index: Interpret ij as raster indices (default is True). If False, assumes ij is coordinates.
+        :type xi: array-like
+        :type yj: array-like
+        :type index: bool
+
+        :returns is_outside: True if ij is outside of the image.
+        """
+        if not index:
+            xi,xj = self.xy2ij(xi,yj)
+
+        if np.any(np.array((xi,yj)) < 0):
+            return True
+        elif xi > self.ds.width or yj > self.ds.height:
+            return True
+        else:
+            return False
+
+    def interp_points(self,pts,nsize=1,mode='linear',band=1):
+
+        """
+        Interpolate raster values at a given point, or sets of points.
+
+       :param pts: Point(s) at which to interpolate raster value. If points fall outside of image,
+       value returned is nan.'
+       :param nsize: Number of neighboring points to include in the interpolation. Default is 1.
+       :param mode: One of 'linear', 'cubic', or 'quintic'. Determines what type of spline is
+           used to interpolate the raster value at each point. For more information, see
+           scipy.interpolate.interp2d. Default is linear.
+       :param band: Raster band to use
+       :type pts: array-like
+       :type nsize: int
+       :type mode: str
+       :type band: int
+
+       :returns rpts: Array of raster value(s) for the given points.
+       :rtype rpts: array-like
+       """
+        assert mode in ['mean', 'linear', 'cubic', 'quintic',
+                        'nearest'], "mode must be mean, linear, cubic, quintic or nearest."
+
+        rpts = []
+
+        #TODO: might need to check if coordinates are center or point in the metadata here...
+
+
+        xx, yy = self.xxyy(offset='center', grid=False)
+        #TODO: right now it's a loop... could add multiprocessing parallel loop outside,
+        # but such a method probably exists already within scipy/other interpolation packages?
+        for pt in pts:
+            i,j = self.xy2ij(pt[0],pt[1])
+            if self.outside_image(i,j, index=True):
+                rpts.append(np.nan)
+                continue
+            else:
+                x = xx[j - nsize:j + nsize + 1]
+                y = yy[i - nsize:i + nsize + 1]
+
+                #TODO: read only that window?
+                z = self.data[band-1, i - nsize:i + nsize + 1, j - nsize:j + nsize + 1]
+                if mode in ['linear', 'cubic', 'quintic', 'nearest']:
+                    X, Y = np.meshgrid(x, y)
+                    try:
+                        zint = griddata((X.flatten(), Y.flatten()), z.flatten(), list(pt), method=mode)[0]
+                    except:
+                        #TODO: currently fails when dealing with the edges
+                        print('Interpolation failed for:')
```

Comment:
Right now the method fails when dealing with edges, returns NaN. An idea for these exception cases?

Reply:
If you were to load the entire data set and use ndimage.map_coordiates, you wouldn't have to worry about your nsize argument and edges.
A better option than loading the entire data set if not desired would be to extract the bounds of the requested pixels, read the array within these bounds and use map_coordinates.